astropy/visualization/scripts/tests/test_fits2bitmap.py

```diff
@@ -19,29 +19,29 @@ def setup_class(self):
         self.array = np.arange(16384).reshape((128, 128))
 
     @pytest.mark.openfiles_ignore
-    def test_function(self, tmpdir):
-        filename = tmpdir.join(self.filename).strpath
+    def test_function(self, tmp_path):
+        filename = str(tmp_path / self.filename)
```

Comment:
Is the str() needed here? If so maybe this is an indication the functions below should be updated to accept Path objects?

Reply:
Are you referring to the test functions or these lines?
```Python
fits.writeto(filename, self.array)
fits2bitmap(filename)
hdulist.writeto(filename)
```

I still use the `str()` so that the output is the same from the original:
```Python
filename = tmpdir.join(self.filename).strpath
# filename='/tmp/pytest-of-jerobado/pytest-176/test_function0/test.fits'
```

Reply:
`io.fits` functions accept pathlib objects, not sure for `fits2bitmap` (which uses matplotlib's `imsave`, probably fine).

Reply:
@jerobado , I think this will be ready once you implement the feedback. Thanks!

Reply:
@jerobado - it's ok to use Path objects now and not convert them to strings if the functions you are testing accept them, so maybe try and remove the ``str()`` around the temporary paths and see if the tests pass?